src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobWorker.cs

```diff
@@ -85,5 +113,38 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
                 }
             }
         }
+
+        public async Task<CancelReindexResponse> Handle(CancelReindexRequest request, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(request, nameof(request));
+
+            return await _retryPolicy.ExecuteAsync(async () =>
+            {
+                using (IScoped<IFhirOperationDataStore> store = _fhirOperationDataStoreFactory())
+                {
+                    ReindexJobWrapper outcome = await store.Value.GetReindexJobByIdAsync(request.JobId, cancellationToken);
+
+                    // If the job is already completed for any reason, return conflict status.
+                    if (outcome.JobRecord.Status.IsFinished())
+                    {
+                        throw new RequestNotValidException(
+                            string.Format(Resources.ReindexJobInCompletedState, outcome.JobRecord.Id, outcome.JobRecord.Status));
+                    }
+
+                    // Try to cancel the job.
+                    outcome.JobRecord.Status = OperationStatus.Canceled;
+                    outcome.JobRecord.CanceledTime = Clock.UtcNow;
```

Comment:
>outcome.JobRecord.CanceledTime = Clock.UtcNow; [](http://example.com/codeflow?start=20&length=46)

Can we have situation where handle of the job and request to cancel job executes on different machines?

Reply:
Excellent point!  I will add some additional checks in the task class.